src/aerie_cli/aerie_client.py

```diff
@@ -1527,16 +1522,24 @@ def upload_scheduling_goals(self, upload_object):
         ]
         """
 
+        # Note that as of Aerie v2.3.0, the metadata (incl. model_id and goal name) are stored in a separate table
+        formatted_upload_objects = []
+        for entry in upload_object:
+            new_entry = {}
+            new_entry["definition"] = entry["definition"]
+            new_entry["metadata"] = {"data": {"name": entry["name"], "models_using": {"data": {"model_id": entry["model_id"]}}}}
+            formatted_upload_objects.append(new_entry)
+        
```

Comment:
For long-term maintainability, would it be better to break the format of `upload_object` than to hard-code this mapping? Who are the users of this method?

Reply:
The only user of this method is `scheduling.upload`. Is it preferable to deliver the `upload_object` to this method as it should be uploaded?

Reply:
I think that's preferable. I'd say we should avoid breaking changes to the API, but I don't know of anyone using the scheduling rule Python API.